### ISSUE TITLE
Not able to debug python function

### ISSUE THREAD

**Issue by Jay1305** [awslabs/aws-sam-cli#1078]
**Description:**
Not able to debug the python function that is out-of-the-box created using SAM. I can debug other python files (not just AWS SAM files) from VS code. I tried to open cmd as Admin (so the VS Code) as well. I might be missing some small thing, but not able to find. 

**Steps to reproduce the issue:**
1. I have followed the guide here - https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/serverless-sam-cli-using-debugging-python.html
2. I modified some part as this guide is not working as exactly written.
*************************************************************************************
1. sam init --runtime python3.6 --name python-debugging
2. cd python-debugging/
3. cd hello_world/
4. pip install -r requirements.txt -t hello_world/build/
5. pip install ptvsd -t hello_world/build/
6. cp app.py hello_world/build/
7. cd ..
8. code . <-- This opened python-debugging workspace
9. Modify hello_world/hello_world/build/app.py with following code
import ptvsd
ptvsd.enable_attach(address=('0.0.0.0', 5890), redirect_output=True)
ptvsd.wait_for_attach()
10. Add Launch.json 
{
    "version": "0.2.0",
    "configurations": [
        {
           "name": "SAM CLI Python Hello World",
           "type": "python",
           "request": "attach",
           "port": 5890,
           "host": "localhost",
           "pathMappings": [
               {
                   "localRoot": "${workspaceFolder}/hello_world/hello_world/build",
                   "remoteRoot": "/var/task"
               }
           ]
       }
   ]
 }
11. sam local start-api -d 5890
12. Start debugger in VS Code with breakpoint. Breakpoint will not hit but I get the JSON output.
*************************************************************************************

**Observed result:**
Breakpoint not hitting
**Expected result:**
Breakpoint should hit

**Current environment:**
Windows 10 Enterprise Build 17134
SAM CLI, version 0.13.0
Docker 18.09.2
Visual Studio Code
AWS SAM 0.10
Python 3.6 32 bit
[Test.zip](https://github.com/awslabs/serverless-application-model/files/3001119/Test.zip)

**Comment by TheSriram** [comment]
https://github.com/awsdocs/aws-sam-developer-guide/issues/8

Let us know if that works for you.

**Comment by Jay1305** [comment]
Thank you Sriram for your reply! I tried installing pyenv on my Windows 10 but it couldn't got installed. I used virtualenv with my python3.6 32 bit version.
The best part is in virtualenv it worked very well!! Thank you! It is still not working without virtual env (but I don't care now :)).
I have 3 different python environments installed under c:\Python with respective directories python3.6, python3.2 and python2.7. The latest one is python3.6 which is in my user Path.
1. Python 3.6 32 bit
2. Python 2.7 32 bit
3. Python 3.2 32 bit
I am able to debug all application and was not able to debug only SAM application, but now I am able to do that using venv. 
Thanks again and sorry for any trouble.
These are in my PATH variable -
C:\Python\Python36-32\Scripts\;C:\Python\Python36-32\;

**Comment by irsali** [comment]
@Jay1305 Could you explain what you have done to resolve this issue or i should say what to do with venv?

**Comment by Jay1305** [comment]
Hey @irsali, I had followed the guide that Sriram sent in the earlier chat. It was correctly working with venv. I setup the venv with python 3.6 environment and then followed what was mentioned by Sriram. Between, it started working without venv too. 
Important note is instead of building the project, I debugged directly from the source which gave me facility to change the code right away. In other words, don't use sam build and instead follow all other steps. Let me know if you see any issues in following Sriram's guide. I can possibly throw some lights.
